stellargraph/data/explorer.py

```diff
@@ -107,6 +81,114 @@ def run(self, **kwargs):
         """
         raise NotImplementedError
 
+    def _raise_error(self, msg):
+        full = "({}) " + msg
+        raise ValueError(full.format(type(self).__name__))
+
+    def _check_parameter_values(self, nodes=None, n=0, length=0, seed=None):
+        """
+        Checks that the parameter values are valid or raises ValueError exceptions with a message indicating the
+        parameter (the first one encountered in the checks) with invalid value.
+
+        Args:
+            nodes: <list> A list of root node ids such that from each node a uniform random walk of up to length l
+            will be generated.
+            n: <int> Number of walks per node id.
+            length: <int> Maximum length of walk measured as the number of edges followed from root node.
+            seed: <int> Random number generator seed
+        Returns:
+            The random state as determined by the seed.
+        """
+        self._check_nodes(nodes)
+        self._check_repetitions(n)
+        self._check_length(length)
+        return self._check_seed(seed)
+
+    def _check_nodes(self, nodes):
+        if nodes is None:
+            self._raise_error("A list of root node IDs was not provided.")
+        if not is_real_iterable(nodes):
+            self._raise_error("Nodes parameter should be an iterable of node IDs.")
+        if (
+            len(nodes) == 0
+        ):  # this is not an error but maybe a warning should be printed to inform the caller
+            print(
+                "({}) WARNING: No root node IDs given. An empty list will be returned as a result.".format(
+                    type(self).__name__
+                )
+            )
+
+    def _check_repetitions(self, n):
+        if type(n) != int:
+            self._raise_error(
+                "The number of walks per root node, n, should be integer type."
+            )
+        if n <= 0:
+            self._raise_error(
+                "The number of walks per root node, n, should be a positive integer."
+            )
+
+    def _check_length(self, length):
+        if type(length) != int:
+            self._raise_error("The walk length, length, should be integer type.")
+        if length <= 0:
+            # Technically, length 0 should be okay, but by consensus is invalid.
+            self._raise_error("The walk length, length, should be a positive integer.")
+
+    def _check_seed(self, seed):
+        if seed is None:
+            # Restore the random state
+            return self._random_state
+        if type(seed) != int:
+            self._raise_error(
+                "The random number generator seed value, seed, should be integer type or None."
+            )
+        if seed < 0:
+            self._raise_error(
+                "The random number generator seed value, seed, should be non-negative integer or None."
+            )
+        # seed the random number generator
+        return random.Random(seed)
```

Comment:
I don't think it's a good idea to have this return a random state: the purpose of this method is checking parameter values, so it should just do that. We can have another method for initialising the random state.

Reply:
I agree, because it breaks the Single Responsibility principle. However, it also removes a lot of explicit code reuse. I could add one more method just for obtaining the random seed.

Reply:
Let's try and stick with the Single Responsibility principle.